gift-certificate-application/service/src/test/java/com/epam/esm/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+        pagination.setSize(SIZE);
+        paginationValidator = mock(PaginationValidatorImpl.class);
+        userList = new ArrayList<>();
+        userList.add(user);
+        userService = new UserServiceImpl(userDao, userValidator, paginationValidator);
+    }
+
+    @Test
+    void testCreate() {
+        User expectedResult = user;
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(true);
+        when(userValidator.isLoginValid(LOGIN)).thenReturn(true);
+        when(userDao.findByLogin(LOGIN)).thenReturn(Optional.empty());
+        when(userDao.create(user)).thenReturn(user);
+        User actualResult = userService.create(user);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testCreateShouldTrowFirstNameFieldValidationException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(false);
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(FieldValidationException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testCreateShouldTrowLastNameFieldValidationException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(false);
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(FieldValidationException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testCreateShouldTrowLoginFieldValidationException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(true);
+        when(userValidator.isLoginValid(LOGIN)).thenReturn(false);
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(FieldValidationException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testCreateShouldTrowUserAlreadyExistException() {
+        when(userValidator.isFirstNameValid(FIST_NAME)).thenReturn(true);
+        when(userValidator.isLastNameValid(LAST_NAME)).thenReturn(true);
+        when(userValidator.isLoginValid(LOGIN)).thenReturn(true);
+        when(userDao.findByLogin(LOGIN)).thenReturn(Optional.of(user));
+        when(userDao.create(user)).thenReturn(user);
+        assertThrows(UserAlreadyExistException.class, () -> userService.create(user));
+    }
+
+    @Test
+    void testFindAll() {
+        List<User> expectedResult = userList;
+        when(userDao.findAll(pagination, User.class)).thenReturn(userList);
+        when(userDao.findEntitiesNumber(User.class)).thenReturn(ENTITIES_NUMBER);
+        when(paginationValidator.validatePagination(pagination, ENTITIES_NUMBER)).thenReturn(pagination);
+        List<User> actualResult = userService.findAll(pagination);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindById() {
+        User expectedResult = user;
```

Comment:
I suppose you can directly use user and do not create an additional variable

Reply:
It's done